tests/phpunit/Authentication/OAuth20Test.php

```diff
@@ -4,55 +4,101 @@
 
 namespace Keboola\GenericExtractor\Tests\Authentication;
 
+use GuzzleHttp\Psr7\Query;
+use GuzzleHttp\Psr7\Uri;
+use GuzzleHttp\Psr7\Utils;
 use Keboola\GenericExtractor\Authentication\OAuth20;
 use Keboola\GenericExtractor\Tests\ExtractorTestCase;
 use Keboola\Juicer\Client\RestClient;
+use Keboola\Juicer\Client\RestRequest;
+use Keboola\Juicer\Tests\HistoryContainer;
+use Keboola\Juicer\Tests\RestClientMockBuilder;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
 use Psr\Log\NullLogger;
 
 class OAuth20Test extends ExtractorTestCase
 {
-    protected function setUp(): void
-    {
-        $this->markTestSkipped('TODO fix test');
-        parent::setUp();
-    }
-
     public function testAuthenticateClientJson(): void
     {
+        // Create OAuth20 auth
         $config = json_decode((string) file_get_contents(__DIR__ . '/../data/oauth20bearer/config.json'), true);
-        $restClient = new RestClient(new NullLogger(), ['base_url' => 'http://example.com'], [], []);
-        $restClient->getClient()->setDefaultOption('headers', ['X-Test' => 'test']);
         $auth = new OAuth20(
-            [],
             $config['authorization'],
             $config['parameters']['api']['authentication']
         );
-        $auth->attachToClient($restClient);
 
-        $request = $restClient->getClient()->createRequest('GET', '/');
-        $restClient->getClient()->send($request);
+        // Create RestClient
+        $history = new HistoryContainer();
+        $restClient = RestClientMockBuilder::create()
+            ->addResponse200('{"foo": "bar1"}')
+            ->addResponse200('{"foo": "bar2"}')
+            ->setGuzzleConfig(['headers' => ['X-Test' => 'test']])
+            ->setHistoryContainer($history)
+            ->setInitCallback(function (RestClient $restClient) use ($auth): void {
+                $auth->attachToClient($restClient);
+            })
+            ->getRestClient();
 
-        self::assertEquals('Bearer testToken', $request->getHeader('Authorization'));
-        self::assertEquals('test', $request->getHeader('X-Test'));
+        // Each request contains Authorization header
```

Comment:
Opat, oba requesty musia obsahovat hlavicku, ... 
V PHP k tomu vela nie je, cele je to definovane v `/data/oauth20bearer/config.json` cez user funkcie.